gui/designwidget.cc

```diff
@@ -265,11 +265,49 @@ void DesignWidget::newContext(Context *ctx)
         std::lock_guard<std::mutex> lock_ui(ctx->ui_mutex);
         std::lock_guard<std::mutex> lock(ctx->mutex);
         
-        getTreeByElementType(ElementType::BEL)->loadContext(ElementType::BEL, ctx);
-        getTreeByElementType(ElementType::WIRE)->loadContext(ElementType::WIRE, ctx);
-        getTreeByElementType(ElementType::PIP)->loadContext(ElementType::PIP, ctx);
-        getTreeByElementType(ElementType::CELL)->loadContext(ElementType::CELL, ctx);
-        getTreeByElementType(ElementType::NET)->loadContext(ElementType::NET, ctx);
+        {
+            std::map<std::pair<int, int>, std::vector<BelId>> belMap;
+            for (auto bel : ctx->getBels()) {

```

Comment:
*auto &bel* ?

Reply:
there's also *const auto &*

Reply:
BelId is supposed to always be small enough to be used as a value not a reference (32 bits for ice40 and 64 bits for ecp5)